evaluation.py

```diff
@@ -4,15 +4,17 @@
 import matplotlib.pyplot as plt
 import numpy as np
 import cv2
+import re
+import os
 
 class Evaluation():
 
-    def __init__(self, model_json, weights, df, num_hg_blocks, batch_size=1):
-        self.model_json = model_json              # json of model to be evaluated
-        self.weights = weights          # weights of model to be evaluated
-        self.df = df                    # df of the the annotations we want
-        self.num_hg_blocks = num_hg_blocks
-        self.batch_size = batch_size
+    def __init__(self, sub_dir, model_json, weights, h_net): 
+        self.model_json = os.path.join(sub_dir,model_json)         # json of model to be evaluated
+        self.weights = os.path.join(sub_dir,weights)          # weights of model to be evaluated
+        self.num_hg_blocks = int(re.match(r'.*stacks_(\d\d)_.*',model_json).group(1))
```

Comment:
We may want to pass in the base directory too if we want to run on all trained models. We can pass in an empty string `''` for `sub_dir` if the model files are saved directly in the model folder. `os.path.join` will omit any strings which are empty. (It doesn't handle `None` though)

```suggestion
    def __init__(self, base_dir, sub_dir, model_json, weights, h_net): 
        self.model_json = os.path.join(base_dir, sub_dir, model_json)         # json of model to be evaluated
        self.weights = os.path.join(base_dir, sub_dir, weights)          # weights of model to be evaluated
        self.num_hg_blocks = int(re.match(r'.*stacks_(\d\d)_.*',model_json).group(1))
```

Reply:
done yo